### ISSUE TITLE
Ask me about SHAR if you'd like

### ISSUE THREAD

**Issue by Bambo-Borris** [Svxy/The-Simpsons-Hit-and-Run#2]
Hello there, 

I've been debugging the SHAR source, and making various debugging edits along the way for over a year now. I have an XP VM image with VS2002 configured to build it (as this is the tooling used at the time). I've also uncovered how to launch the various tools the developers created, one even being their debug variables editing tool. For any of those curious feel free to drop some questions on the topic of how to get going in building the source, or what areas are intereting. I'm a regularly over in the Simpson's Hit & Run Speedrun Discord server discussing how various glitches work. 

I don't claim to know it all. This source leak is big, and there's a lot of understanding of how the game works was built by Donut Team & aldelaro5 prior to the source leak. I know a slice, but I'm more than happy to share what I can with the community. I love the game, it was a part of my childhood and I'm glad to help those who also want to understand it or just understand how a game can be engineered in C++ for multiple platforms.

Thanks,
Bambo.

**Comment by LWSS** [comment]
It's missing the audio from what I can see. Same with hulk. Makes it a bit more of a pita to figure out how to bake the sounds

**Comment by Bambo-Borris** [comment]
> It's missing the audio from what I can see. Same with hulk. Makes it a bit more of a pita to figure out how to bake the sounds

Are you referring to unprocessed/raw audio assets being missing from the leak?

**Comment by LWSS** [comment]
![image](https://github.com/Svxy/The-Simpsons-Hit-and-Run/assets/8827133/8bfa4bfb-b796-474a-92d5-2557cc1e2b4d)

yes there should be a sounds/ folder (this is Hulk but same scenario) that contains sounds/scripts to bake .rms files

**Comment by Bambo-Borris** [comment]
> ![image](https://user-images.githubusercontent.com/8827133/255431604-8bfa4bfb-b796-474a-92d5-2557cc1e2b4d.png)
> 
> yes there should be a sounds/ folder (this is Hulk but same scenario) that contains sounds/scripts to bake .rms files

Yeah I haven't taken much look into the audio baking process, but missing the raw files is a pretty killer problem here. Is `radmusic` one of their tools that's got it's own VS project in the source leak, do you know?

**Comment by floatas** [comment]
Hey @Bambo-Borris what are your thoughts about custom missions ? is it possible to add them, is there any editor ?

**Comment by Bambo-Borris** [comment]
> Hey @Bambo-Borris what are your thoughts about custom missions ? is it possible to add them, is there any editor ?

It's probably easier to look at doing that via mods to the original game (check out Donut Team's website). 

There's scope to do it in the source leak, but remember, it's a buggy and unstable mess of a build. You can't use the mod launcher with the executables built by the source leak. So you can't gain stability fixes. There's also no FPS cap. There's also bugs that aren't present on the official release because the source code dates between the last console release and the pc release. Meaning it's still 'in development' state. 

So yeah you could add missions to a build from source, but whether you'd want to is a totally different matter.

**Comment by cursedUs64-git** [comment]
hey umm how can i build shr for windows 10? i have found 2 tutorials, one on github and other on youtube. Tho, i get a error about the shader files (.vsh) when it is compiling. Can anyone help me here?

**Comment by Bambo-Borris** [comment]
> hey umm how can i build shr for windows 10? i have found 2 tutorials, one on github and other on youtube. Tho, i get a error about the shader files (.vsh) when it is compiling. Can anyone help me here?
> 

Not sure, I build it on the original VS2002 compiler to keep aligned with the original release for testing purposes. The SHAR solution doesn't trivially upgrade to modern VS either. Maybe installing the 2010 DirectX SDK will help with shader stuff, it certainly did for my XP build VM.

**Comment by cursedUs64-git** [comment]
> > hey umm how can i build shr for windows 10? i have found 2 tutorials, one on github and other on youtube. Tho, i get a error about the shader files (.vsh) when it is compiling. Can anyone help me here?
> 
> Not sure, I build it on the original VS2002 compiler to keep aligned with the original release for testing purposes. The SHAR solution doesn't trivially upgrade to modern VS either. Maybe installing the 2010 DirectX SDK will help with shader stuff, it certainly did for my XP build VM.

it still shows errors about the shaders (.vsh format), even if i installed directx 2010 sdk, as shown in this video: https://www.youtube.com/watch?v=stIWQAM_UM4
Still, i get a error, like .vsxproj file somethin, idk

**Comment by Svxy** [comment]
> it still shows errors about the shaders (.vsh format), even if i installed directx 2010 sdk, as shown in this video: https://www.youtube.com/watch?v=stIWQAM_UM4 Still, i get a error, like .vsxproj file somethin, idk

Tried using an earlier sdk???

**Comment by 3UR** [comment]
hello @Bambo-Borris any idea how to use TheWatcher for SRR2? (Debug tool from radcore)
